### PR TITLE
Project Proposal <yr942> <xl3725>

### PR DESCRIPTION
**Pull request by li928904115**

### CODE REVIEW

File: yr942-xl3725/project.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  - Yafu Ruan <yr942@nyu.edu>
+  - Xuan Li <xl3725@nyu.edu>
+---
+
+> In what follows instructions (including this one) are in blockquotes, while
+> text not in blockquotes is a sample. You should of course replace the title
+> author metadata that shows up above with appropriate values.
+>
+> You should start your proposal by stating a problem statement or an overall
+> goal  as below.
```
Comment (apanda) [inline yr942-xl3725/project.md]: Can you delete the instructions in blockquote.

File: yr942-xl3725/project.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+> resources) you plan to use, how you plan to make progress, and how you plan to
+> evaluate your view.
+
+We plan to construct a Reinforcement Learning model based information from Kubernetes API and job requirements.
+
+* Firstly we design the basic training algorithm using pytorch.
+* Next, we will collect resource information from Kubernetes API and transform them to vectors in a proper way. We combine it with job information, which will also be a vector, and take it as the state variable.
+* After that, we will design the reward metric to determine how “good” the decision is made by the model. The metric needs to consider the trade-off between the overall utilization and the balance of resources and clusters.
+
+We plan to run our experiments on multiple CloudLab nodes.
```
Comment (apanda) [inline yr942-xl3725/project.md:26]: What workloads do you plan to use to evaluate your scheduler?

File: yr942-xl3725/project.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+> You should also talk about how you will evaluate your progress, and what you
+> think the ideal end goal would look like.
+
+We will evaluate our work by comparing the performance (average utilization of each cluster) of our scheduling policy with some traditional policy, like Least Load, Round Robin and Random.
```
Comment (apanda) [inline yr942-xl3725/project.md]: Do you plan to compare to all three policies, or which ones?

**Review by apanda — COMMENTED** [review]
The proposal looks reasonable, I have some questions that you should answer.

File: yr942-xl3725/project.md (hunk 2 above)
Comment (YafuR) [reply yr942-xl3725/project.md:26]: In the training process, we are going to simulate job loads, whose arrival time, duration and recourses come from certain distributions. This should be similar to the work load simulated in DeepRM as described below(https://people.csail.mit.edu/alizadeh/papers/deeprm-hotnets16.pdf):

“jobs arrive online according to a Bernoulli process. The average job arrival rate is chosen such that the average load varies between 10% to 190% of cluster capacity. We assume two resources, i.e., with capacity {1r, 1r}. Job durations and resource demands are chosen as follows: 80% of the jobs have duration uniformly chosen between 1t and 3t; the remaining are chosen uniformly from 10t to 15t. Each job has a dominant resource which is picked independently at random. The demand for the dominant resource is chosen uniformly between 0.25r and 0.5r and the demand of the other resource is chosen uniformly between 0.05r and 0.1r. “

In the testing process, we will try to write a python program to do almost the same and run on k8s(do not sure if it is feasible right now, we will figure it).

File: yr942-xl3725/project.md (hunk 3 above)
Comment (li928904115) [reply yr942-xl3725/project.md]: We will first compare it with Random and Least Load, if we still have time, we will also compare it with round robin.

File: yr942-xl3725/project.md (hunk 2 above)
Comment (apanda) [reply yr942-xl3725/project.md:26]: Can you add the information in the comment above to the proposal itself.

File: yr942-xl3725/project.md (hunk 3 above)
Comment (apanda) [reply yr942-xl3725/project.md]: Add this information to the proposal itself.

**Review by apanda — APPROVED** [review]
Thanks for adding this additional information.

**Pull request merged**